spec/serializers/search_and_compare/course_serializer_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'rails_helper'
+
+describe SearchAndCompare::CourseSerializer do
+  describe 'json output' do
+    let(:resource) { serialize(course, serializer_class: described_class) }
+
+    subject { resource }
+
+    context 'an existing course' do
+      let(:course_factory_args) do
+        {
+          provider: provider,
+          accrediting_provider: accrediting_provider,
+          name: expected_json[:Name],
+          course_code: expected_json[:ProgrammeCode],
+          start_date: expected_json[:StartDate],
+        }
+      end
+
+      let(:course) do
+        create :course, **course_factory_args
+      end
+
+      let(:provider) do
+        create :provider,
+               provider_code: expected_json[:Provider][:ProviderCode],
+               provider_name: expected_json[:Provider][:Name]
+      end
+      let(:accrediting_provider) do
+        create :provider,
+               provider_code: expected_json[:AccreditingProvider][:ProviderCode],
+               provider_name: expected_json[:AccreditingProvider][:Name]
```

Comment:
I'd say don't use `expected_json` here and just hard-code the values.

Reply:
will do in next set of PRs